symengine/derivative.cpp

```diff
@@ -697,6 +697,61 @@ void DiffVisitor::bvisit(const Tuple &self)
     throw SymEngineException("Derivative doesn't exist.");
 }
 
+void DiffVisitor::bvisit(const IdentityMatrix &self)
+{
+    throw SymEngineException("Derivative doesn't exist.");
```

Comment:
Isn't this just the ZeroMatrix?

Reply:
Is it ok to keep derivatives outside the scope of this PR?

Reply:
Sure, but the error message needs to be changed then.

Reply:
Ah, now I see what you mean. It should signal not implemented.

Reply:
Changed to `NotImplemented`